app/src/common/shared/org/mozilla/vrbrowser/ui/keyboards/ChineseKeyboard.java

```diff
@@ -0,0 +1,376 @@
+package org.mozilla.vrbrowser.ui.keyboards;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.util.Log;
+
+import com.readystatesoftware.sqliteasset.SQLiteAssetHelper;
+
+import org.mozilla.vrbrowser.R;
+import org.mozilla.vrbrowser.input.CustomKeyboard;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class ChineseKeyboard implements KeyboardInterface {
```

Comment:
I would consider to change this file name to ChinesePinyinKeyboard or just PinyinKeyboard because we will support `ChineseZhuyinKeyboard` in the early future.